examples/keycloak-pepr/capabilities/keycloak-api.ts

```diff
@@ -0,0 +1,264 @@
+import { generateRandomPassword } from './util';
+import { fetch } from "pepr";
+export { keycloakAPI };
+
+class keycloakAPI {
+  baseURL: string;
+  connected: boolean;
+  clientId: string;
+  username: string;
+  password: string;
+  accessToken: string;
+
+  constructor(baseURL: string, password: string, clientId: string) {
+    this.baseURL = baseURL
+    this.connected = false
+    this.password = password
+    this.clientId = clientId
+    this.username = 'user'
+  }
+
+  // makes the connection to the keycloak API and gets an accessToken
+  private async connect() {
+    if (this.connected == true) {
+      return
+    }
+
+    interface TokenResponse {
+      access_token: string
+    }
+
+    const tokenUrl = `${this.baseURL}/realms/master/protocol/openid-connect/token`;
+
+    const requestData = {
+      'grant_type': 'password',
+      'client_id': 'admin-cli',
+      'username': this.username,
+      'password': this.password
+    }
+
+    var { data, ok, status, statusText } = await fetch<TokenResponse>(tokenUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      body: new URLSearchParams(requestData),
+    });
+    if (!ok) {
+      throw new Error(`${status} ${statusText} from tokenUrl`)
+    }
+    this.accessToken = data.access_token;
+    this.connected = true
+  }
+
+
+  // return the tokenized headers
+  getHeaders() {
+    return {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${this.accessToken}`,
+    }
+  }
+
+  // 
+  async createUser(realmName: string, username: string, email: string, firstName: string, lastName: string): Promise<string> {
+    await this.connect()
```

Comment:
calling `this` in the manner is considered dangerous. In almost all cases the use of `this` only makes sense when used with a class and typically an arrow function to guarantee `this` context safety (which can move in javascript)